doc/lua_api.md

```diff
@@ -8540,8 +8540,17 @@ child will follow movement and rotation of that bone.
     * Passing no arguments resets lighting to its default values.
     * `light_definition` is a table with the following optional fields:
       * `saturation` sets the saturation (vividness; default: `1.0`).
-        * values > 1 increase the saturation
-        * values in [0,1] decrease the saturation
+        * It is applied according to the function `result = b*(1-s) + c*s`, where:
+          * `c` is the original color
+          * `b` is the greyscale version of the color with the same luma
+          * `s` is the saturation set here
+        * The resulting color always has the same luma (perceived brightness) as the original.
+        * This means that:
+          * value of 1 (default) doesn't change the saturation
+          * values > 1 increase the saturation
+          * values between 0 and 1 decrease the saturation
+          * value of 0 makes the world greyscale
+          * values < 0 alter the hue severely, causing an effect similar to inversion, but keeping original luma and being symmetrical in terms of saturation (eg. -1 and 1 is the same saturation and luma, but different hues)
```

Comment:
1. Line limited to < 95 characters (according to https://dev.minetest.net/Code_style_guidelines#Don.27t_let_things_get_too_large )
2. More compact description (debatable)

```suggestion
          * A value of 0 corresponds to greyscale, and > 1 to oversaturated.
          * values < 0 cause an effect similar to inversion, but keeping original luma and
            being symmetrical in terms of saturation
            (e.g. -1 and 1 result in the same saturation and luma, but with different hues)
```

Reply:
You're not including the `0 < saturation < 1` case though (nor the default 1, although that may be considered implicit).

Reply:
Also I'm pretty sure I've seen longer lines like that in markdown files here.

Reply:
>  although that may be considered implicit

I'd consider that common sense. Though that does not mean that I'm correct with this assumption.

> Also I'm pretty sure I've seen longer lines like that in markdown files here.

The C++ side does sometimes not follow the guidelines either, but changes that touch such lines are expected to correct it to provide a more uniform appearance.

Reply:
I wrote something similar to your suggestion. See if it's good enough.